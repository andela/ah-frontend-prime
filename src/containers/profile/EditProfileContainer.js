import React, { Component } from "react";
import EditProfile from "../../components/userProfile/EditProfile";

export class EditProfileContainer extends Component {
  render() {
    return (
      <div>
        <EditProfile />
      </div>
    );
  }
}

export default EditProfileContainer;
