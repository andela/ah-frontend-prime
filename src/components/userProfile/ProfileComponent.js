import React from "react";
import "../../styles/profile.scss";
import { Link } from "react-router-dom";

const ProfileComponent = props => {
  const { full_name, bio, image, username } = props;
  return (
    <div>
      <div className="container-personal-profile">
        <div>
          <div className="user-details">
            <div>
              <h3>{`${username}'s profile`}</h3>
            </div>
            <div>
              <Link to="/editprofile" className="btn btn-outline-success">
                {" "}
                edit profile
              </Link>
            </div>
            <br />
            <div>
              <p>{`Fullname: ${full_name}`}</p>
              <p>{`bio: ${bio}`}</p>
            </div>
            <div>
              <span>Following:&nbsp;</span>
              <span>
                <i>19</i>
              </span>
              <span>&nbsp;|&nbsp;</span>
              <span>Followers:&nbsp;</span>
              <span>
                <i>19</i>
              </span>
            </div>
          </div>
          <div className="img-container">
            <img
              src={image}
              style={{ borderRadius: "50%", width: "200px", height: "200px" }}
            />
          </div>
        </div>
      </div>
    </div>
  );
};
export default ProfileComponent;
