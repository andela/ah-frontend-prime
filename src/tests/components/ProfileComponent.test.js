import { shallow } from "enzyme";
import React from "react";
import ProfileComponent from "../../components/userProfile/ProfileComponent";
import EditProfileContainer from "../../containers/profile/EditProfileContainer";

describe("profileComponent tests", () => {
  const props = {
    onSubmit: jest.fn(),
    errors: null,
    generalError: ""
  };
  it("should render without crushing", () => {
    const wrapper = shallow(<ProfileComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("edit profile container tests", () => {
  const props = {
    onSubmit: jest.fn(),
    image: "",
    bio: "",
    full_name: ""
  };
  it("should render without crushing", () => {
    const wrapper = shallow(<EditProfileContainer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
