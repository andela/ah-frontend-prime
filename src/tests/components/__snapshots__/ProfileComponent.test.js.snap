// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit profile container tests should render without crushing 1`] = `ShallowWrapper {}`;

exports[`profileComponent tests should render without crushing 1`] = `ShallowWrapper {}`;
